R/seroprevalence_data.R

```diff
@@ -120,4 +120,203 @@ prepare_bin_data <- function(serodata) {
+}
+
+#' Function that randomly generates a sample of counts for a simulated dataset
+#'
+#' @param sim_data A dataframe object containing the following columns:
+#' \tabular{ll}{
+#' \code{birth_year} \tab List of years in which the subjects were borned \cr \tab \cr
+#' \code{tsur} \tab Year of the survey\cr \tab \cr
+#' \code{country} \tab Default to 'none'.\cr \tab \cr
+#' \code{survey} \tab Survey label \cr \tab \cr
+#' \code{age_mean_f} \tab Age \cr \tab \cr
+#' }
+#' @param foi Numeric atomic vector corresponding to the desired Force-of-Infection.
+#' @param size_age_class Size of each age group specified by either an atomic
+#' vector of the same size as \code{foi} or an integer.
+#' This corresponds to the number of trials \code{size} in \link[stats]{rbinom}.
+#' @param seed The seed for random number generation.
+#' @return A simulated list of counts following a binomial distribution in accordance with a given 
+#' force of infection and age class sizes.
+#' @examples
+#'\dontrun{
+#'
+#' }
+#' @export
+get_sim_counts <- function(sim_data, foi, size_age_class, seed = 1234) {
+  sim_probabilities <- get_sim_prob(sim_data = sim_data, foi = foi, seed = 1234)
+  sim_counts <- purrr::map_int(sim_probabilities, ~rbinom(1, size_age_class, .))
+
+  return(sim_counts)
+}
+
+#' Function that generates simulated positive counts assuming a known historical force of infection
+#'
+#' @param foi Numeric atomic vector corresponding to the desired Force-of-Infection
+#' @param size_age_class Size of each age group specified by either an atomic
+#' vector of the same size as \code{foi} or an integer.
+#' This corresponds to the number of trials \code{size} in \link[stats]{rbinom}.
+#' @return Dataframe object containing the simulated data generated from \code{foi}
+#' @examples
+#'\dontrun{
+#' size_age_class = 5
+#' foi <- rep(0.02, 50)
+#' sim_data <- generate_sim_data(foi = foi,
+#'                               size_age_class = size_age_class,
+#'                               tsur = 2050,
+#'                               birth_year_min = 2000,
+#'                               survey_label = 'sim_constant_foi')
+#' }
+#' @export
+generate_sim_data <- function(foi,
+                              size_age_class,
+                              tsur,
+                              birth_year_min,
+                              survey_label,
+                              test = "fake",
+                              antibody = "IgG",
+                              seed = 1234
+                              ){
+    sim_data <- data.frame(birth_year = c(birth_year_min:(tsur - 1))) %>%
+        mutate(tsur = tsur,
+            country = 'None',
+            test = test,
+            antibody = antibody,
+            survey = survey_label,
+            age_mean_f = tsur - birth_year)
+    sim_data <- sim_data %>%
+        mutate(counts = get_sim_counts(sim_data, foi, size_age_class, seed = seed),
+            total = size_age_class) %>%
```

Comment:
I would change this to be `n_tested` to indicate a) that this is a number and b) to indicate what the total refers to.

Reply:
Same as above (see #112 )